src/Raku/ast/package.rakumod

```diff
@@ -337,12 +337,16 @@ class RakuAST::Package
         if $!declarator eq 'role' {
             $type.HOW.set_body_block($type, $!body.meta-object);
 
+            # The role needs to be composed before we add the possibility to the group
+            $type.HOW.compose($type);
+
             my $group := $!role-group;
-            $group.HOW.add_possibility($group, $type) unless $group =:= Mu;
+            $group.HOW.add_possibility($group, $type) unless $group =:= Mu || $group =:= $type;
```

Comment:
In what case would $group be the same as $type?

Reply:
Good catch. This was logic ported from `Perl6/Actions`. I just double checked and it passes the same number of spectests without this change.